extensions/ql-vscode/src/remote-queries/run-remote-query.ts

```diff
@@ -262,98 +257,7 @@ export async function prepareRemoteQueryRun(
-    throw new Error(`Variant analysis is not supported by this version of CodeQL. Please upgrade to v${cli.CliVersionConstraint.CLI_VERSION_REMOTE_QUERIES
-      } or later.`);
-  }
-
-  const {
-    actionBranch,
-    base64Pack,
-    repoSelection,
-    queryFile,
-    queryMetadata,
-    controllerRepo,
-    queryStartTime,
-    language,
-  } = await prepareRemoteQueryRun(cliServer, credentials, uri, progress, token);
-
-  if (isVariantAnalysisLiveResultsEnabled()) {
-    const queryName = getQueryName(queryMetadata, queryFile);
-    const variantAnalysisLanguage = parseVariantAnalysisQueryLanguage(language);
-    if (variantAnalysisLanguage === undefined) {
-      throw new UserCancellationException(`Found unsupported language: ${language}`);
-    }
-
-    const queryText = await fs.readFile(queryFile, 'utf8');
-
-    const variantAnalysisSubmission: VariantAnalysisSubmission = {
-      startTime: queryStartTime,
-      actionRepoRef: actionBranch,
-      controllerRepoId: controllerRepo.id,
-      query: {
-        name: queryName,
-        filePath: queryFile,
-        pack: base64Pack,
-        language: variantAnalysisLanguage,
-        text: queryText,
-      },
-      databases: {
-        repositories: repoSelection.repositories,
-        repositoryLists: repoSelection.repositoryLists,
-        repositoryOwners: repoSelection.owners
-      }
-    };
-
-    const variantAnalysisResponse = await ghApiClient.submitVariantAnalysis(
-      credentials,
-      variantAnalysisSubmission
-    );
-
-    const processedVariantAnalysis = processVariantAnalysis(variantAnalysisSubmission, variantAnalysisResponse);
-
-    await variantAnalysisManager.onVariantAnalysisSubmitted(processedVariantAnalysis);
-
-    void logger.log(`Variant analysis:\n${JSON.stringify(processedVariantAnalysis, null, 2)}`);
-
-    void showAndLogInformationMessage(`Variant analysis ${processedVariantAnalysis.query.name} submitted for processing`);
-
-    void commands.executeCommand('codeQL.openVariantAnalysisView', processedVariantAnalysis.id);
-    void commands.executeCommand('codeQL.monitorVariantAnalysis', processedVariantAnalysis);
-
-    return { variantAnalysis: processedVariantAnalysis };
-  } else {
-    const apiResponse = await runRemoteQueriesApiRequest(credentials, actionBranch, language, repoSelection, controllerRepo, base64Pack);
-
-    if (!apiResponse) {
-      return;
-    }
-
-    const workflowRunId = apiResponse.workflow_run_id;
-    const repositoryCount = apiResponse.repositories_queried.length;
-    const remoteQuery = await buildRemoteQueryEntity(
-      queryFile,
-      queryMetadata,
-      controllerRepo,
-      queryStartTime,
-      workflowRunId,
-      language,
-      repositoryCount);
-
-    // don't return the path because it has been deleted
-    return { query: remoteQuery };
-  }
-}
-
-async function runRemoteQueriesApiRequest(
+export async function runRemoteQueriesApiRequest(
```

Comment:
Can this and `buildRemoteQueryEntity` be moved to the `RemoteQueriesManager` too?

Reply:
Yes, these can be moved out of this file. I'd prefer to not place them into the `RemoteQueriesManager` itself since it doesn't depend on the state of that class, but I'll move the `buildRemoteQueryEntity` into `remote-query.ts` and create a new file for this method in `remote-queries-api.ts`.